### PR TITLE
MIscellaneous python typing fixes (III)

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: lms/models/_mixins/public_id.py [hunk 1]
```diff
@@ -63,5 +63,5 @@ def public_id(self) -> str | None:
         )
 
     @public_id.comparator
-    def public_id(self):
```
Comment (marcospri) [inline lms/models/_mixins/public_id.py:66]: The decorator takes cares of calling the right method, the name doesn't matter.

File: lms/models/event.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Type(str, Enum):
         GRADE = "grade"
         ERROR_CODE = "error_code"
 
-    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    id = mapped_column(sa.Integer, autoincrement=True, primary_key=True)
```
Comment (marcospri) [inline lms/models/event.py:23]: These fixes errors elsewhere, they seem to provide better type inference via SQLAlchemy.

File: lms/services/canvas_api/client.py [hunk 3]
```diff
@@ -170,6 +170,9 @@ def course_sections(self, course_id):
     class _CourseSectionsSchema(RequestsResponseSchema, _SectionSchema):
         """Schema for the "list course sections" responses."""
 
+        class Meta:
```
Comment (marcospri) [inline lms/services/canvas_api/client.py:173]: Overriding with the same value as `_SectionSchema` to make mypy happy.

File: lms/services/canvas_studio.py [hunk 4]
```diff
@@ -67,7 +67,7 @@ def post_load(self, data, **_kwargs):
 
 class APICallInfo(TypedDict):
     path: str
-    authUrl: str | None
+    authUrl: NotRequired[str]
```
Comment (marcospri) [inline lms/services/canvas_studio.py:70]: Not required (ie not dict key) vs optional (key can be None)

File: lms/services/group_info.py [hunk 5]
```diff
@@ -46,7 +46,7 @@ def upsert_group_info(self, grouping: Grouping, params: dict):
         # AI should be many:many, and we reflect that wrongness here.
         group_info.application_instance = grouping.application_instance
 
-        group_info.type = self._GROUPING_TYPES[grouping.type]
+        group_info.type = self._GROUPING_TYPES[grouping.type]  # type: ignore
```
Comment (marcospri) [inline lms/services/group_info.py:49]: We don't really maintain group_info that much. THe fix here is to refactor it completely or remove it.

File: lms/services/rsa_key.py [hunk 6]
```diff
@@ -121,7 +121,7 @@ def _as_jwk_public_key(rsa_key) -> dict:
             encoding=serialization.Encoding.PEM,
             format=serialization.PublicFormat.SubjectPublicKeyInfo,
         )
-        return jwk.RSAKey(
+        return jwk.RSAKey(  # type: ignore
```
Comment (marcospri) [inline lms/services/rsa_key.py:124]: Not sure about this one, ignoring.

File: lms/tasks/email_digests.py [hunk 7]
```diff
@@ -144,12 +144,12 @@ def send_instructor_email_digest(
     """
     # Caution: datetime.fromisoformat() doesn't support all ISO 8601 strings!
     # This only works for the subset of ISO 8601 produced by datetime.isoformat().
-    created_before: datetime = datetime.fromisoformat(created_before)
+    created_before: datetime = datetime.fromisoformat(created_before)  # type: ignore
```
Comment (marcospri) [inline lms/tasks/email_digests.py:147]: Redefining the variable with differnt type.

File: lms/views/admin/_schemas.py [hunk 8]
```diff
@@ -19,8 +19,8 @@ def deserialize(self, value, attr, data, **kwargs):
         # pylint:disable=compare-to-empty-string
         if value == missing or value.strip() == "":
             return None
-        return super().deserialize(value, attr, data, **kwargs)
+        return super().deserialize(value, attr, data, **kwargs)  # type: ignore
```
Comment (marcospri) [inline lms/views/admin/_schemas.py:22]: Not entirely sure about this one either.

File: pyproject.toml [hunk 9]
```diff
@@ -149,21 +149,15 @@ disable_error_code = [
...
-    "index",
-    "list-item",
-    "misc",
-    "no-redef",
-    "no-untyped-call",
-    "no-untyped-def",
-    "return-value",
 ]
 
-plugins = []
```
Comment (marcospri) [inline pyproject.toml:161]: No need for this with no plugins.

File: pyproject.toml [hunk 10]
```diff
@@ -149,21 +149,11 @@ disable_error_code = [
...
-
 [[tool.mypy.overrides]]
-module="tests.*"
```
Comment (marcospri) [inline pyproject.toml:164]: Combine these two.

File: lms/services/canvas_studio.py (hunk 4 above)
Comment (acelaya) [reply lms/services/canvas_studio.py:70]: This reminds me to `undefined` vs `null` in Javascript 😂

**Review by acelaya — APPROVED** [review]

**Pull request merged**